lib/order.rb

```diff
@@ -0,0 +1,104 @@
+require_relative 'Menu'
+require_relative 'Texter'
+class Order
+  attr_reader :pricelist
+
+  def initialize(set_menu: Menu)
+    @pricelist = set_menu.new
+  end
+  def place_order
+    intro
+    notepad = take_order
+    check_total(notepad)? gucci : miscalulated
+  end
+
+  protected
+
+  def take_order(dishes: 0, pad: [])
+    phone_order = gets.chomp
+    while !phone_order.empty? do
+      request = phone_order.split(', ')
+      volume = request[1].to_i
+      pad.push({request[0] => volume})
+      dishes += volume
+      puts "So that's #{dishes} plates so far"
+      phone_order = gets.chomp
+    end
+    pad
+  end
+
+  def gucci
+    puts "looks good to me! I'll just send you a text with the delivery estimate"
+    puts "what number would you like me to send it to?"
+    puts "please input your number plus country code, e.g. 44789085764"
+    number = gets.chomp
+    latest_delivery = Time.new + 1800
+    self.send_confirmation(number: number, time: latest_delivery)
+  end
+
+  def miscalulated
+     puts "nah sorry bad math there, do you want to try again?"
+     puts "Y[es]/N[o]?"
+     choice = gets.chomp
+     if choice == "Y"
+       place_order
+     else
+       puts "no worries then have a great day."
+     end
+  end
+
+  def intro
+    puts "Please place your order by typing the name of each dish followed by a comma and the quantity per dish"
+    puts "once you've given the full order let us know roughly how much you think it should be, we'll check and then take your number."
+    puts "for example, 'Carbonara, 2'\n \n 14'"
+  end
+
+  def send_confirmation(method:Texter, number:, time:)
+     while check_number(number) do
+       puts "ah that number doesn't have the right number of digits, please re-enter"
+       number = gets.chomp
+     end
+    method.new(num: number, text_time: time).make_the_call
+  end
+  def check_number(number)
+    num_digits(number) != 11
+  end
+  def num_digits(num)
+    Math.log10(num).to_i + 1
+  end
```

Comment:
Seems like this method also collects a phone number. the gucci method also did this, perhaps a bit of shared responsibility leakage?